### PR TITLE
Convert more doctests into unittests.

### PR DESCRIPTION
**Pull request by hannosch**
Also use `unittest.skipIf/Unless` to skip tests, so the test count is the same in all test combinations.

Unitests make up ~50% of the tests now, up from ~30% :)

### CODE REVIEW

File: src/Acquisition/tests.py [hunk 1]
```diff
@@ -3685,12 +3453,21 @@ def test_suite():
         unittest.makeSuite(TestCreatingWrappers),
         unittest.makeSuite(TestPickle),
         unittest.makeSuite(TestInterfaces),
+        unittest.makeSuite(TestMixin),
```
Comment (jamadden) [inline src/Acquisition/tests.py]: May I suggest removing all of the explicit calls to `makeSuite(SomeTestClass)` and replacing them with a single `unittest.defaultTestLoader.loadTestsFromName(__name__)`? That makes adding new test classes less error prone.

**Comment by hannosch** [comment]
> We don't use coveralls.io on Travis CI yet here. Did you happen to run the tox coverage environment before and after to verify that coverage stays the same?

Yes. I did a before/after comparison and coverage stayed at exactly the same value (some 98.91%). The uncovered lines being reported also stayed the same, so I felt confident committing the change.

File: src/Acquisition/tests.py (hunk 1 above)
Comment (hannosch) [reply src/Acquisition/tests.py]: Excellent idea, added in second commit

**Comment by hannosch** [comment]
> Unfortunately the diffs for this sort of thing are painful to try to read (I know I've subjected others to that)...

Same here, I'm not anticipating anyone to read through the diff line-by-line. I know I wouldn't with this sort of change.

**Comment by hannosch** [comment]
Thanks for the review :)

The remaining doctests all depend more heavily on printing or the `show` helper, to print ASCII-art tree structures. My motivation ran out for today ;)

**Pull request merged**